src/components/DetailsOverlay/Toggle/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import { palette } from "@leafygreen-ui/palette";
+import Toggle from "@leafygreen-ui/toggle";
+import { Disclaimer } from "@leafygreen-ui/typography";
+import Cookie from "js-cookie";
+import { FORMAT_LOG_V2, PRETTY_PRINT_BOOKMARKS } from "constants/cookies";
+import { QueryParams } from "constants/queryParams";
+import { size } from "constants/tokens";
+import { useQueryParam } from "hooks/useQueryParam";
+import { FilterRow, StyledSubtitle } from "../styles";
+
+const { gray } = palette;
+
+interface BaseToggleProps {
+  ["data-cy"]?: string;
+  leftLabel?: string;
+  rightLabel?: string;
+  label: string;
+  value: boolean;
+  onChange: (checked: boolean) => void;
+}
+
+const BaseToggle: React.FC<BaseToggleProps> = ({
+  "data-cy": dataCy,
+  rightLabel = "ON",
+  leftLabel = "OFF",
+  label,
+  value,
+  onChange,
+}) => (
+  <FilterRow>
+    <StyledSubtitle> {label} </StyledSubtitle>
+    <ToggleWithLabel>
+      <StyledDisclaimer>{leftLabel}</StyledDisclaimer>
+      <Toggle
+        aria-labelledby="base-toggle"
+        checked={value}
+        data-cy={dataCy}
+        onChange={onChange}
+        size="small"
+      />
+      <StyledDisclaimer>{rightLabel}</StyledDisclaimer>
+    </ToggleWithLabel>
+  </FilterRow>
+);
```

Comment:
Maybe this should be its own component?  Some Storybook stories and tests would be nice. :)